packages/vc-handler-ld-local/src/ld-credential-module.ts

```diff
@@ -0,0 +1,167 @@
+import { purposes } from '@digitalcredentials/jsonld-signatures'
+import * as vc from '@digitalcredentials/vc'
+import { CredentialIssuancePurpose } from '@digitalcredentials/vc'
+import { VerifiableCredentialSP, VerifiablePresentationSP } from '@sphereon/ssi-sdk-core'
+import {
+  CredentialPayload,
+  IAgentContext,
+  IKey,
+  IKeyManager,
+  IResolver,
+  PresentationPayload,
+  VerifiableCredential,
+  VerifiablePresentation,
+} from '@veramo/core'
+import Debug from 'debug'
+
+import { LdContextLoader } from './ld-context-loader'
+import { LdDocumentLoader } from './ld-document-loader'
+import { LdSuiteLoader } from './ld-suite-loader'
+
+export type RequiredAgentMethods = IResolver & Pick<IKeyManager, 'keyManagerGet' | 'keyManagerSign'>
+
+const ProofPurpose = purposes.ProofPurpose
+const AssertionProofPurpose = purposes.AssertionProofPurpose
+const AuthenticationProofPurpose = purposes.AuthenticationProofPurpose
+
+const debug = Debug('veramo:w3c:ld-credential-module-local')
+
+export class LdCredentialModule {
+  /**
+   * TODO: General Implementation Notes
+   * - (SOLVED) EcdsaSecp256k1Signature2019 (Signature) and EcdsaSecp256k1VerificationKey2019 (Key)
+   * are not useable right now, since they are not able to work with blockChainId and ECRecover.
+   * - DID Fragment Resolution.
+   * - Key Manager and Verification Methods: Veramo currently implements no link between those.
+   */
+
+  ldSuiteLoader: LdSuiteLoader
+  private ldDocumentLoader: LdDocumentLoader
+
+  constructor(options: { ldContextLoader: LdContextLoader; ldSuiteLoader: LdSuiteLoader }) {
+    this.ldSuiteLoader = options.ldSuiteLoader
+    this.ldDocumentLoader = new LdDocumentLoader(options)
+  }
+
+  async issueLDVerifiableCredential(
+    credential: CredentialPayload,
+    issuerDid: string,
+    key: IKey,
+    verificationMethodId: string,
+    purpose: typeof ProofPurpose = new CredentialIssuancePurpose(),
+    context: IAgentContext<RequiredAgentMethods>
+  ): Promise<VerifiableCredentialSP> {
+    // fixme: We need to look at the verificationMethod in meta of the key, and then look up the suiteloader, as for instance ed25519 can be the 2018 or 2020 version
+    const suite = this.ldSuiteLoader.getSignatureSuiteForKeyType(key.type, key.meta?.verificationMethod?.type)
```

Comment:
Will it be implemented in this ticket?

Reply:
It was already fixed, but not removed here